code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/identity/TokenResourceManager.java

```diff
@@ -83,19 +89,35 @@ private Token createToken(ApiRequest request) {
         token.setIdentities(transFormedIdentities);
         token.setUserIdentity(identityManager.untransform(token.getUserIdentity(), true));
         token.setJwt(authTokenDao.createToken(token.getJwt(), token.getAuthProvider(), ((Policy) ApiContext.getContext().getPolicy()).getAccountId()).getKey());
+        AuthToken authToken= authTokenDao.getTokenByKey(token.getJwt());
+        String tokenId = (String)ApiContext.getContext().getIdFormatter().formatId(objectManager.getType(Token.class), authToken.getId());
```

Comment:
Hoping we can get rid of this, but FYI it looks like you are double obfuscating the token:
```
curl 'http://192.168.42.5:8080/v2-beta/token' -H 'Origin: http://192.168.42.5:8080' -H 'Accept-Encoding: gzip, deflate' -H 'Accept-Language: en-US,en;q=0.8' -H 'x-api-action-links: actionLinks' -H 'content-type: application/json' -H 'x-api-no-challenge: true' -H 'accept: application/json'  --data-binary '{"code":"craig:craig","authProvider":"localauthconfig"}' | jq .
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   587  100   532  100    55  31080   3213 --:--:-- --:--:-- --:--:-- 31294
{
  "id": "1t1t10",  // <------------------------------------------------ Not right
  "type": "token",
  "links": {
    "self": "http://192.168.42.5:8080/v2-beta/tokens/1t1t10"
  },
  "baseType": "token",
  "actionLinks": {},
  "accountId": "1a1",
  "authProvider": "localAuthConfig",
  "code": null,
  "enabled": true,
  "jwt": "XkaFQJvDLpjntwrk54pbk4JYEvudHHqkmMZWMSy6",
  ... 
}
```